cupy/core/_kernel.pyx

```diff
@@ -701,28 +706,42 @@ cdef inline bint _check_should_use_min_scalar(list in_args) except? -1:
             max_array_kind >= max_scalar_kind)
 
 
-cdef tuple _guess_routine(name, dict cache, list ops, list in_args, dtype):
+cdef dict _mst_unsigned_to_signed = {
+    i: (numpy.iinfo(j).max, (i, j))
+    for i, j in [(numpy.dtype(i).type, numpy.dtype(i.lower()).type)
+                 for i in "BHILQ"]}
+cdef _numpy_min_scalar_type = numpy.min_scalar_type
+
+cdef _min_scalar_type(x):
```

Comment:
Could you add a comment to the function?  For example,
"A non-negative integer may have two locally minimum scalar types: signed/unsigned integer. Return both for `can_cast`, while `numpy.min_scalar_type` only returns the unsigned type."

Reply:
Thanks